CalculadoraDos/Controllers/ButtonController.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  ButtonController.swift
+//  CalculadoraDos
+//
+//  Created by Luis Fernando Sanchez Muñoz on 9/4/24.
+//
+
+import Foundation
+import UIKit
+
+class ButtonController {
+    var pulsations: [CalculatorKeys] = []
+    func addCharacter(character: CalculatorKeys) {
+        guard !pulsations.isEmpty,
+              case .number(let num2) = character,
+              case .number(let num1) = pulsations[pulsations.count-1]  else {
+            pulsations.append(character)
+            return
+        }
+        guard let rawInt = Int(("\(num1)\(num2)")) else { return }
+        pulsations[pulsations.count-1] = CalculatorKeys.number(rawInt)
+    }
```

Comment:
Si evitamos el uso de -1 nos garantizamos que no nos salimos del array

```suggestion
    func addCharacter(character: CalculatorKeys) {
        if case .number(let newKey) = character,
           case .number(let last) = pulsations.last {
            pulsations.removeLast()
            pulsations.append(.number(10 * last + newKey))
        } else {
            pulsations.append(character)
        }
    }
```